### PR TITLE
get_route_to() now returns all vrfs

### PR DESCRIPTION
**Pull request by kbirkeland**
Added test case for multi-vrf environment

<!-- Make sure you have read http://napalm.readthedocs.io/en/latest/contributing/index.html --!>

### CODE REVIEW

File: napalm_eos/eos.py [hunk 1]
```diff
@@ -1003,120 +1003,133 @@ def get_mac_address_table(self):
...
-                            'local_preference': local_preference,
-                            'local_as': local_as,
-                            'remote_as': remote_as,
-                            'remote_address': remote_address,
-                            'preference2': preference2,
-                            'communities': communities
-                        }
-                    })
-                    routes[prefix].append(bgp_route)
+            command = 'show ip{ipv} route vrf {_vrf} {destination} {protocol} detail'.format(
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: Please add this command in a list of commands, then execute all of them in a single request.
Then iterate through the complete reply.

**Review by mirceaulinic — CHANGES_REQUESTED** [review]
Hi @kbirkeland - thanks for this PR.

File: napalm_eos/eos.py [hunk 2]
```diff
@@ -1003,120 +1003,133 @@ def get_mac_address_table(self):
...
+        # Placeholder for vrf arg
+        vrf = ''
+
+        # Right not iterating through vrfs is necessary
+        # show ipv6 route doesn't support vrf 'all'
+        if vrf == '':
+            vrfs = self.get_network_instances().keys()
+        else:
+            vrfs = [vrf]
+
```
Comment (dbarrosop) [inline napalm_eos/eos.py]: You are hardcoding the variable `vrf` so the `if..else` statement seems a bit unnecessary :)

File: napalm_eos/eos.py [hunk 3]
```diff
@@ -1003,120 +1003,133 @@ def get_mac_address_table(self):
     def get_route_to(self, destination='', protocol=''):
         routes = {}
 
+        # Placeholder for vrf arg
+        vrf = ''
+
+        # Right not iterating through vrfs is necessary
+        # show ipv6 route doesn't support vrf 'all'
+        if vrf == '':
+            vrfs = self.get_network_instances().keys()
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: Can we find a different way to get the list of VRFs on the device? `get_network_instances` is already pretty heavy and we don't want this method to depend on it.

File: napalm_eos/eos.py (hunk 2 above)
Comment (kbirkeland) [reply napalm_eos/eos.py]: It is, but it makes the implementation of a vrf argument a simple change.  I can remove that if you want.

File: napalm_eos/eos.py (hunk 3 above)
Comment (kbirkeland) [reply napalm_eos/eos.py]: See ca2c4f3a912eb127bfd6c7ce0e3d6d79f16209b0

File: napalm_eos/eos.py (hunk 1 above)
Comment (kbirkeland) [reply napalm_eos/eos.py]: See ae695b469b7028dd098ea1704911b15654d97c69

**Review by mirceaulinic — APPROVED** [review]
Also this one looks good to me. I'll put it on hold till next week when I need to test it before getting this in. Thanks!

**Comment by mirceaulinic** [comment]
Thanks @kbirkeland

**Pull request merged**